### PR TITLE
improved error messages

### PR DESCRIPTION
**Pull request by Felalolf**
Improves error messages for channels.

### CODE REVIEW

File: src/main/scala/viper/gobra/reporting/VerifierError.scala [hunk 1]
```diff
@@ -263,6 +286,11 @@ case class InsufficientPermissionError(info: Source.Verifier.Info) extends Verif
   override def message: String = s"permission to ${info.origin.tag.trim} might not suffice"
 }
 
+case class InsufficientPermissionError2(tag: String) extends VerificationErrorReason {
```
Comment (jcp19) [inline src/main/scala/viper/gobra/reporting/VerifierError.scala:289]: Would probably rename it to something else such as `InsufficientPermissionToTagError`

File: src/main/scala/viper/gobra/translator/util/ViperWriter.scala [hunk 2]
```diff
@@ -370,6 +370,42 @@ object ViperWriter {
       } yield ()
     }
 
+    /* Emits Viper statements. */
+    def assertWithDefaultReason(cond: vpr.Exp, error: Source.Verifier.Info => VerificationError): Writer[Unit] = {
```
Comment (ArquintL) [inline src/main/scala/viper/gobra/translator/util/ViperWriter.scala:374]: `assertWithDefaultReason` and `exhaleWithDefault` are currently unused, aren't they?

**Review by ArquintL — APPROVED** [review]
Only a minor question to double-check that you're aware of this

File: src/main/scala/viper/gobra/translator/util/ViperWriter.scala [hunk 3]
```diff
@@ -370,6 +370,42 @@ object ViperWriter {
...
+        _ <- write(res)
+        _ <- errorT({
+          case e@ vprerr.ExhaleFailed(Source(info), reason, _) if e causedBy res =>
+            reasonT(info, reason)
+        })
+      } yield ()
+    }
+
+    /* Emits Viper statements. */
+    def exhaleWithDefault(cond: vpr.Exp, error: Source.Verifier.Info => VerificationError): Writer[Unit] = {
```
Comment (jcp19) [inline src/main/scala/viper/gobra/translator/util/ViperWriter.scala:398]: would rename it to `exhaleWithDefaultReason` to be consistent with `assertWithDefaultReason`

**Review by jcp19 — APPROVED** [review]

File: src/main/scala/viper/gobra/translator/util/ViperWriter.scala (hunk 2 above)
Comment (Felalolf) [reply src/main/scala/viper/gobra/translator/util/ViperWriter.scala:374]: Yes , they are unused, but I think that they are useful.

**Pull request merged**